src/badge/Badge.d.ts

```diff
@@ -0,0 +1,14 @@
+import { PureComponent, ReactChildren } from 'react';
+
+declare namespace Badge {
+
+  type Props = {
+    label: ReactChildren,
```

Comment:
Typescript's `ReactChildren` is quite different from `tcomb`'s one
I'm pretty sure this is not enough...

I tried this quickly in a project and it complains:
```ts
type X = {
    child: React.ReactChildren
  };

    const x: X = {
      child: <div />
    };

    const x: X = {
      child: ''
    };
```